freeipa/src/main/java/com/sequenceiq/freeipa/sync/ProviderChecker.java

```diff
@@ -31,39 +32,48 @@
     @Inject
     private StackInstanceProviderChecker stackInstanceProviderChecker;
 
+    @Inject
+    private FlowLogService flowLogService;
+
     @Value("${freeipa.autosync.update.status:true}")
     private boolean updateStatus;
 
     public List<ProviderSyncResult> updateAndGetStatuses(Stack stack, Set<InstanceMetaData> checkableInstances) {
         return checkedMeasure(() -> {
             List<ProviderSyncResult> results = new ArrayList<>();
             List<CloudVmInstanceStatus> statuses = stackInstanceProviderChecker.checkStatus(stack, checkableInstances);
-            statuses.forEach(s -> {
-                Optional<InstanceMetaData> instanceMetaData = checkableInstances.stream()
-                        .filter(i -> s.getCloudInstance().getInstanceId().equals(i.getInstanceId()))
-                        .findFirst();
-                if (instanceMetaData.isPresent()) {
-                    InstanceStatus instanceStatus = updateStatuses(s, instanceMetaData.get());
-                    if (instanceStatus != null) {
-                        results.add(new ProviderSyncResult("", instanceStatus, false, s.getCloudInstance().getInstanceId()));
+            if (flowLogService.isOtherFlowRunning(stack.getId())) {
```

Comment:
Don't need this here, we check this at the start

Reply:
please see the above comments